math/src/msm/pippenger.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{
+    cyclic_group::IsCyclicBilinearGroup,
+    field::{element::FieldElement, traits::IsLinearField},
+};
+
+use super::MSM;
+
+pub struct Pippenger {
+    window_size: usize,
+}
+
+/// Calculates the multiscalar multiplication using the Pippenger algorithm
+/// as described in section 2.3 of the paper "cuZK: Accelerating Zero-Knowledge
+/// Proof with A Faster Parallel Multi-Scalar Multiplication Algorithm on GPUs".
+/// Reference: https://eprint.iacr.org/2022/1321.pdf
+impl Pippenger {
+    pub fn new(window_size: usize) -> Self {
+        assert!(window_size > 0, "window size should be greater than zero");
```

Comment:
This function should return a `Result`instead of using an `assert` because doing the latter would panic in runtime. 

If you don't have any choice but to panic (this is not the case tho) it is better to describe the cases in which that function would panic and why.

Reply:
@ilitteri, I changed to return a `Result` now.